Gender/pronoun update

saraht45 opened the pull request:
Removed hard-coded gender/pronoun options. By default, male/female/neuter are still there. More options can be added in the TextIF:Gender woe object. The gender drop down list in tree of woe can be updated to reflect these options by modifying the DTD:DTD woe object.

noahgibbs commented:
This looks really good. I'm checking for stuff you may have missed, so generally poking around...

I *hope* the ancient and horrible Bilbo code in skoot/data/vault/Generic/DB/RefreshDB.xml is now unused. But you can find the string "male" there and it setting the UrBody in a way that looks really bad. So I assume it's not used any more? Ditto for skoot/data/vault/Generic/DB/UrPeople/refresh.xml, which looks like basically the same code.

I'm not 100% sure how the Merry/Thing-level gender that includes nonbinary interacts with the lower-level DGD-style gender which is a small integer. Do you know how the base:gender property is set? I ask because there's still, for instance, skoot/data/vault/Lib/echo.xml, which checks that number and declares a voice male or female...

I'm going to ignore various forms like StartStoryFrame - I figure you have whatever start story we're actually using covered. Similarly, I'm not going to worry about various StoryBuilder/StoryCoder/etc type scripts. That kind of thing is usually easy enough to fix after the fact or when needed.

Do we want to just get rid of the emitallmale/emitallfemale commands? I don't see them being used anywhere and I'm not sure when they would be.

Huh. In skoot/usr/SkotOS/sys/npc-bodies.c it has separate "drag" destinations for male and female, used in set_teleport. My knowledge of NPC bodies isn't good enough to know what that's for. But it seems to effectively mean "male" and "anybody not definitively male" right now. If it's unused that may not matter.

But in general this looks really good. What I'm seeing left over is little stuff. Well done.

saraht45 commented:
> I _hope_ the ancient and horrible Bilbo code in skoot/data/vault/Generic/DB/RefreshDB.xml is now unused. But you can find the string "male" there and it setting the UrBody in a way that looks really bad. So I assume it's not used any more? Ditto for skoot/data/vault/Generic/DB/UrPeople/refresh.xml, which looks like basically the same code.

Everything in skoot/data/vault/Generic/ should probably go away because the majority of it has bilbo. I'll look through this and make changes in a different PR.

> I'm not 100% sure how the Merry/Thing-level gender that includes nonbinary interacts with the lower-level DGD-style gender which is a small integer. Do you know how the base:gender property is set? I ask because there's still, for instance, skoot/data/vault/Lib/echo.xml, which checks that number and declares a voice male or female...

I updated skoot/data/vault/Lib/echo.xml to only broadcast a "voice" as opposed to a male or female voice. This is the bit of code that handles how shouts are seen in adjacent rooms ("You hear a male voice shouting from the east, 'Doh!'"). Developers can decide how they want to handle this by modifying this woe object.

> I'm going to ignore various forms like StartStoryFrame - I figure you have whatever start story we're actually using covered. Similarly, I'm not going to worry about various StoryBuilder/StoryCoder/etc type scripts. That kind of thing is usually easy enough to fix after the fact or when needed.

Correct. A new start story will take care of this.

> Do we want to just get rid of the emitallmale/emitallfemale commands? I don't see them being used anywhere and I'm not sure when they would be.

These can go away.

> Huh. In skoot/usr/SkotOS/sys/npc-bodies.c it has separate "drag" destinations for male and female, used in set_teleport. My knowledge of NPC bodies isn't good enough to know what that's for. But it seems to effectively mean "male" and "anybody not definitively male" right now. If it's unused that may not matter.

I'm not sure this drag code is necessary. If it is, it should be added at the Merry level.


I have added these changes to the existing PR :)

noahgibbs commented:
Cool. Looks good to me.
